packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -310,7 +310,7 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
       description: `Subnets for ${id} database`,
       vpc: props.instanceProps.vpc,
       vpcSubnets: props.instanceProps.vpcSubnets,
-      removalPolicy: props.removalPolicy === RemovalPolicy.RETAIN ? props.removalPolicy : undefined,
+      removalPolicy: props.removalPolicy,
```

Comment:
Why this change? A `SubnetGroup` is not a stateful resource I think? Or is it?

Also, pretty sure this doesn't work for `props.removalPolicy` equal to `SNAPSHOT`.

Reply:
> Why this change?

This change was just to simplify.

This used to only set a removalPolicy if the value was `RETAIN`, and not otherwise (so it would filter out `SNAPSHOT`, basically).

Kinda don't care for that behavior. I'd rather have CloudFormation yell at a user for choosing `SNAPSHOT` than for us to silently filter that out and implicitly defaulting to `DELETE`.

Reply:
I don't understand, sorry.

`SNAPSHOT` is a valid removal policy to set for the Cluster. But it's not a valid removal policy to set for `SubnetGroup`. So you're actually breaking a valid customer scenario with this change (that worked fine before).

Reply:
This is still broken.